packages/shared/package.json

```diff
@@ -1,21 +1,24 @@
 {
-  "name": "shared",
+  "name": "@supabase/auth-helpers-shared",
   "version": "0.1.0",
-  "private": true,
-  "main": "index.ts",
+  "main": "dist/index.js",
+  "module": "dist/index.mjs",
+  "types": "dist/index.d.ts",
+  "publishConfig": {
+    "access": "public"
+  },
   "files": [
-    "adapters/*",
-    "utils/*",
-    "types.ts"
+    "dist"
   ],
+  "scripts": {
+    "build": "tsup"
+  },
   "devDependencies": {
     "@supabase/supabase-js": "^1.35.3",
     "next": "^12.1.5",
     "react": ">=17.0.2 <18.0.0 || >=18.0.0-0 <19.0.0",
-    "react-dom": "^17.0.2 || ^18.0.0-0"
-  },
-  "peerDependencies": {
-    "react": ">=17.0.2 <18.0.0 || >=18.0.0-0 <19.0.0",
-    "react-dom": "^17.0.2 || ^18.0.0-0"
+    "react-dom": "^17.0.2 || ^18.0.0-0",
+    "tsconfig": "workspace:*",
+    "tsup": "^5.12.5"
```

Comment:
With this setup we should be able to move the `jwtDecoder` back into shared and have `jose` be a dependency here, right?